src/uct/tcp/tcp_iface.c

```diff
@@ -118,8 +119,7 @@ static ucs_status_t uct_tcp_iface_query(uct_iface_h tl_iface, uct_iface_attr_t *
     attr->cap.am.max_bcopy = am_buf_size;
     attr->cap.am.max_iov   = iface->config.zcopy.max_iov;
 
-    if ((attr->cap.am.max_iov > 0) &&
-        (iface->config.tx_seg_size >= sizeof(uct_tcp_ep_zcopy_ctx_t))) {
+    if (attr->cap.am.max_iov > UCT_TCP_EP_AM_ZCOPY_SERVICE_IOV_COUNT) {
```

Comment:
shouldn't we set attr->cap.am.max_iov  to iface->config.zcopy.max_iov-UCT_TCP_EP_AM_ZCOPY_SERVICE_IOV_COUNT?